### ISSUE TITLE
get_beam_directions_grid is broken for triclinic

### ISSUE THREAD

**Issue by din14970** [pyxem/diffsims#155]
**Describe the bug**
`get_beam_directions_grid` needs to be looked over once more. It looks fine for most crystal systems, except the triclinic one. Now this is making me doubt whether it's working correctly for the other ones, so I will have to verify.

**To Reproduce**
```python
grid_tric = get_beam_directions_grid("triclinic", 3)

from orix.quaternion.rotation import Rotation
from orix.vector.vector3d import Vector3d

rotations_regular =  Rotation.from_euler(np.deg2rad(i))
rot_reg_test = rotations_regular*Vector3d.zvector()

from orix.plot.stereographic_plot import StereographicPlot
fig, ax = plt.subplots(figsize = (20, 10), ncols=2, subplot_kw=dict(projection="stereographic"))

ax[0].scatter(rot_reg_test, c="C0")  # blue
ax[0].show_hemisphere_label(color="C0")  # Pass keyword arguments to text()
ax[0].set_labels(zlabel=None)

ax[1].hemisphere = "lower"  # /"south", or "upper"/"north"
ax[1].scatter(rot_reg_test, c="C1")  # orange
ax[1].show_hemisphere_label()
```
![afbeelding](https://user-images.githubusercontent.com/25320842/109864464-29cf7380-7c63-11eb-81f8-e529d24f2ce6.png)

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem here. eg) The use of an uncommon system/version

**Comment by hakonanes** [comment]
With this code with diffsims v0.4 and orix v0.6.dev0

```python
from diffsims import generators
import matplotlib.pyplot as plt
import numpy as np
from orix import plot, quaternion, vector

plt.rcParams["axes.grid"] = True

grid_tric = generators.rotation_list_generators.get_beam_directions_grid(
    "triclinic", 3
)
rotations_regular = quaternion.Rotation.from_euler(np.rad2deg(grid_tric))
rot_reg_test = rotations_regular * vector.Vector3d.zvector()
fig, ax = plt.subplots(figsize=(10, 5), ncols=2, subplot_kw=dict(projection="stereographic"))
ax[0].scatter(rot_reg_test, s=2, c="C0")
ax[0].set_labels(zlabel=None)
ax[0].show_hemisphere_label()
ax[1].hemisphere = "lower"
ax[1].scatter(rot_reg_test, s=2, c="C1")
ax[1].show_hemisphere_label()
ax[1].set_labels(zlabel=None)
fig.savefig("/home/hakon/triclinic_sampling.png", bbox_inches="tight")
```

I get this figure

![triclinic_sampling](https://user-images.githubusercontent.com/12139781/109874198-15917380-7c6f-11eb-9935-9a7847b52850.png)

I believe the trick was using deg2rad instead of rad2deg?

**Comment by hakonanes** [comment]
Is this more like expected?

**Comment by din14970** [comment]
No, one shoulf really get a regular almost equally spaced grid. It works for all the other crystal systems. I'm sure its a small oversight and I'll try to fix it tomorrow

**Comment by pc494** [comment]
It's my screw up I think, in the function block we have:

```python
    if crystal_system == "triclinic":
        return points_in_cartesians # see comment below

    # crop to stereographic triangle which depends on crystal system
    corners = crystal_system_dictionary[crystal_system]
    a, b, c = corners[0], corners[1], corners[2]
    if len(a) == 4:
        a, b, c = uvtw_to_uvw(a), uvtw_to_uvw(b), uvtw_to_uvw(c)


    # eliminates those points that lie outside of the stereographic triangle
    epsilon = -1e-13
    points_in_cartesians = points_in_cartesians[
        np.dot(np.cross(a, b), c) * np.dot(np.cross(a, b), points_in_cartesians.T)
        >= epsilon
    ]
    points_in_cartesians = points_in_cartesians[
        np.dot(np.cross(b, c), a) * np.dot(np.cross(b, c), points_in_cartesians.T)
        >= epsilon
    ]
    points_in_cartesians = points_in_cartesians[
        np.dot(np.cross(c, a), b) * np.dot(np.cross(c, a), points_in_cartesians.T)
        >= epsilon
    ]


    angle_grid = beam_directions_grid_to_euler(points_in_cartesians) #correct for all other systems
    return angle_grid
```

Seems clear to me the the early return statement (it was lazy of me to write that, granted) needs to be:
`return beam_directions_grid_to_euler(points_in_cartesians)`
Not sure how I missed that at the time...

**Comment by hakonanes** [comment]
yeah, and of course from_euler takes radians, not degrees, so my rad2deg use is wrong as well...

**Comment by din14970** [comment]
So it seems we just need to put `beam_directions_grid_to_euler` around the `points_in_cartesian` return statement.

**Comment by din14970** [comment]
Went through the list and it seems that monoclinic is also incorrect. The point list:
```
"monoclinic": [(0, -1, 0), (0, 0, 1), (0, 1, 0)],
```

Which corresponds to 1/4 of the sphere if I am not mistaken.

The image:
![afbeelding](https://user-images.githubusercontent.com/25320842/109945418-83718580-7cd7-11eb-84f0-58709cc8dfbc.png)

This probably results from the fact that the points are linearly dependent. So for monoclinic we need a special case.

**Comment by pc494** [comment]
Yup, at a glance I agree with you, the cross product between (0,-1,0) and  (0,1,0) isn't going to behave.

**Comment by pc494** [comment]
Closed by #156
